src/spaces/discrete.jl

```diff
@@ -273,3 +273,35 @@ function move_agent_single!(
     move_agent!(agent, position, model)
     return agent
 end
+
+"""
+    swap_agents!(agent1, agent2, model::ABM{<:DiscreteSpace})
+
+Swaps agents function used for swapping the postion of two agents.
+"""
+function swap_agents!(agent1, agent2, model::ABM{<:DiscreteSpace})
+    agents_no_pos = []
+
+    pos_a = agent1.pos    
+    pos_b = agent2.pos
+
+    if(isnothing(pos_a))
+        push!(agents_no_pos,agent1)
+    end 
+
+    if(isnothing(pos_b))
+        push!(agents_no_pos,agent2)
+    end 
+
+    if (pos_a == pos_b)
+        return 
+    end
+
+    random_agent_from_space!(agent1, model)
+    random_agent_from_space!(agent2, model)
+
+    add_agent_to_space!(agent1, model, pos_a)    
```

Comment:
`add_agent_to_space!` doesn't accept a `pos` value as its third argument. Instead, you should set `agent1`'s pos to `pos_b` and `agent2`'s to `pos_a`, and then call `add_agent_to_space!` with just two arguments for each agent.